migrations/sql/V2018.02.20.1505__AddUserGroup.sql

```diff
@@ -0,0 +1,50 @@
+CREATE TABLE user_group (
+  name        TEXT NOT NULL,
+  description TEXT NULL,
+  PRIMARY KEY (name)
+);
+
+CREATE TABLE user_group_user (
+  username   TEXT NOT NULL,
+  user_group TEXT NOT NULL,
+  PRIMARY KEY (username, user_group)
+);
+
+ALTER TABLE user_group_user
+  ADD FOREIGN KEY (username) REFERENCES app_user (username);
+ALTER TABLE user_group_user
+  ADD FOREIGN KEY (user_group) REFERENCES user_group (name);
+
+CREATE TABLE user_group_role (
+user_group TEXT NOT NULL ,
+role INTEGER ,
+scope_id TEXT NOT NULL ,
+PRIMARY KEY (user_group, role, scope_id)
+);
+
+ALTER TABLE user_group_role ADD FOREIGN KEY (user_group) REFERENCES user_group (name);
+ALTER TABLE user_group_role ADD FOREIGN KEY (role) REFERENCES role (id);
+
+-- create standard groups
+INSERT INTO user_group (name, description)
+VALUES ('GAVI', 'All members of GAVI'),
+  ('Gates', 'All members of the Gates foundation'),
+  ('Modellers', 'Members of modelling groups'),
+  ('Secretariat', 'Members of the consortium secretariat');
```

Comment:
Part of me wants these internal primary keys to be all lower case? But I don't know I have a good reason for it.

Reply:
Wait, I do have a good reason: Our API spec requires IDs in URLs to be `lowercase.with.dots`. So these should be too.